### PR TITLE
CLI: fix --help view when in context of invalid service

### PR DESCRIPTION
**Pull request by medikoo**
Closes: #7913 

Additionally it covers:
- Ensure to show version info unconditionally if `-v` or `--version` is passed
- Ensure to show interactlive CLI help unconditionally if `--help-interactive` is passed
- Fix issues in config resolution error handling
  - Ensure operational errors are thrown in recognized cases
  - Validate config object type independently of used file type (so far it was validated only if it was JS or TS)
- Upgrade to use `@serverless/test` at v4
- Rewrite `lib/classes/Serverless.test.js` tests to rely on `runServerless` util (so they do not depend on internals, and it's safe to pursue refactors which do not change the outcome)

### CODE REVIEW

**Comment by yugasun** [comment]
That's great~

File: lib/classes/CLI.js [hunk 1]
```diff
@@ -63,30 +65,40 @@ class CLI {
     this.log = function() {};
   }
 
+  isHelpRequest(processedInput) {
+    const commands = processedInput.commands;
+    const options = processedInput.options;
+
+    if (options.version || options.v) return true;
```
Comment (hkbarton) [inline lib/classes/CLI.js:72]: should we check `commands.length === 1 && commands[0] === 'version'` as well? I saw we're checking this below? maybe have a function `isVersionRequest` ?

Comment (medikoo) [reply lib/classes/CLI.js:72]: > Should we check `commands.length === 1 && commands[0] === 'version'` as well? 

We're checking that at L76.

> maybe have a function `isVersionRequest `

I think it's fine to cover both _help_ and _version_ under one function. I didn't try to name it as `isHelpOrVersionRequest`, for simplicity

**Comment by rmykh** [comment]
Guys, did you miss that `-v` in all the help files still says it is equal to `--verbose`? Now all the CI processes that deployed anything with serverless in verbose mode like `serverless deploy --stage test -v` are going to just show version of serverless instead of actual deploy.
This is important. @medikoo

**Comment by jameswoodley** [comment]
This has broken deploys with the -v flag. See #7944

**Pull request merged**